solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -1331,12 +1331,24 @@ public ConfigData getSecurityProps(boolean getFresh) {
    * Returns the baseURL corresponding to a given node's nodeName -- NOTE: does not (currently)
    * imply that the nodeName (or resulting baseURL) exists in the cluster.
    *
-   * @lucene.experimental
+   * @param nodeName name of the node
+   * @return url that looks like {@code https://localhost:8983/solr}
    */
   public String getBaseUrlForNodeName(final String nodeName) {
     return Utils.getBaseUrlForNodeName(nodeName, getClusterProperty(URL_SCHEME, "http"));
   }
 
+  /**
+   * Returns the V2 baseURL corresponding to a given node's nodeName -- NOTE: does not (currently)
+   * imply that the nodeName (or resulting baseURL) exists in the cluster.
+   *
+   * @param nodeName name of the node
+   * @return url that looks like {@code https://localhost:8983/api}
+   */
+  public String getBaseUrlV2ForNodeName(final String nodeName) {
+    return Utils.getBaseUrlForNodeName(nodeName, getClusterProperty(URL_SCHEME, "http"), true);
```

Comment:
is it just me, or does this method look exactly like the one in `getBaseUrlForNodeName`???   Where do we do the `/api` conversion?

Reply:
Looking up to line 1338

Reply:
It's actually a different overload of the method which accepts `boolean isV2` parameter. But I agree - perhaps it's more clear to also use the same overload in line 1338, and pass `isV2 = false` explicitly.

https://github.com/apache/solr/blob/3837eebcd73e103c083d7b02bdf6904d08fc34b5/solr/solrj/src/java/org/apache/solr/common/util/Utils.java#L719-L738

Reply:
yeah...  Or maybe some javadocs to explain it?

Reply:
Let me know when you decide what you want to do here, and then i look forward to merging it!

Reply:
I added javadoc to the methods in the Utils class, and refactored the code a little in 50c959f

Reply:
thanks!